specs/_features/eip7732/fork-choice.md

```diff
@@ -260,185 +262,151 @@ def get_checkpoint_block(store: Store, root: Root, epoch: Epoch) -> Root:
+            if not state.validators[i].slashed
+        ]
+        attestation_score = Gwei(
+            sum(
+                state.validators[i].effective_balance
+                for i in unslashed_and_active_indices
+                if (
+                    i in store.latest_messages
+                    and i not in store.equivocating_indices
+                    and is_supporting_vote(store, node, store.latest_messages[i])
+                )
+            )
+        )
 
-### New `compute_reveal_boost`
+        if store.proposer_boost_root == Root():
+            # Return only attestation score if ``proposer_boost_root`` is not set
+            return attestation_score
 
-This is a similar helper to the last two, the only difference is that the reveal
-boost is only applied to the full version of the node when querying for the same
-slot as the revealed payload.
+        # Calculate proposer score if ``proposer_boost_root`` is set
+        proposer_score = Gwei(0)
 
-```python
-def compute_reveal_boost(store: Store, state: BeaconState, node: ChildNode) -> Gwei:
-    if store.payload_reveal_boost_root == Root():
-        return Gwei(0)
-    ancestor = get_ancestor(store, store.payload_reveal_boost_root, node.slot)
-    if ancestor.root != node.root:
-        return Gwei(0)
-    if node.slot >= store.blocks[store.payload_reveal_boost_root].slot:
-        ancestor.is_payload_present = True
-    if ancestor.is_payload_present != node.is_payload_present:
-        return Gwei(0)
-    committee_weight = get_total_active_balance(state) // SLOTS_PER_EPOCH
-    return (committee_weight * PAYLOAD_REVEAL_BOOST) // 100
-```
+        # ``proposer_boost_root`` is treated as a vote for the
+        # proposer's block, in the current slot. Proposer boost
+        # is applied accordingly, to all ancestors
+        message = LatestMessage(
+            slot=get_current_slot(store),
+            root=store.proposer_boost_root,
+            payload_present=False,
+        )
+        if is_supporting_vote(store, node, message):
+            proposer_score = get_proposer_score(store)
 
-### Modified `get_weight`
+        return attestation_score + proposer_score
+```
 
-*Note*: `get_weight` is modified to only count votes for descending chains that
-support the status of a triple `Root, Slot, bool`, where the `bool` indicates if
-the block was full or not. `Slot` is needed for a correct implementation of
-`(Block, Slot)` voting.
+### New `get_fork_choice_children`
 
 ```python
-def get_weight(store: Store, node: ChildNode) -> Gwei:
-    state = store.checkpoint_states[store.justified_checkpoint]
-    unslashed_and_active_indices = [
-        i
-        for i in get_active_validator_indices(state, get_current_epoch(state))
-        if not state.validators[i].slashed
-    ]
-    attestation_score = Gwei(
-        sum(
-            state.validators[i].effective_balance
-            for i in unslashed_and_active_indices
+def get_fork_choice_children(
```

Comment:
Other functions are named like `get_forkchoice_store` so probably this one should be `get_forkchoice_children`, or probably `get_node_children`.